src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
-                break;
+    package com.shipmentEvents.handlers;
+    
+    import java.nio.charset.StandardCharsets;
+    import java.time.Duration;
+    import java.util.ArrayList;
+    import java.util.HashMap;
+    import java.util.Iterator;
+    import java.util.List;
+    import java.util.Map;
+    import java.util.Map.Entry;
+    
+    import javax.crypto.Cipher;
+    import javax.crypto.SecretKey;
+    import javax.crypto.spec.SecretKeySpec;
+    
+    import com.amazonaws.regions.Regions;
+    import com.amazonaws.services.lambda.runtime.Context;
+    import com.amazonaws.services.lambda.runtime.RequestHandler;
+    import com.amazonaws.services.lambda.runtime.LambdaLogger;
+    import com.amazonaws.services.lambda.runtime.events.ScheduledEvent;
+    import com.amazonaws.services.s3.AmazonS3;
+    import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+    import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+    import com.amazonaws.services.s3.model.DeleteObjectsRequest.KeyVersion;
+    import com.amazonaws.services.s3.model.ObjectListing;
+    import com.amazonaws.services.s3.model.S3ObjectSummary;
+    import com.shipmentEvents.util.Constants;
+    import com.shopify.ShopifySdk;
+    import com.shopify.model.ShopifyShop;
+    
+    import org.apache.commons.lang3.tuple.MutablePair;
+    import org.apache.commons.lang3.tuple.Pair;
+    
+    
+    public class EventHandler implements RequestHandler<ScheduledEvent, String> {
+    
+        /**
+         * Shipment events for a carrier are uploaded to separate S3 buckets based on the source of events. E.g., events originating from
+         * the hand-held scanner are stored in a separate bucket than the ones from mobile App. The Lambda processes events from multiple
+         * sources and updates the latest status of the package in a summary S3 bucket every 15 minutes.
+         * 
+         * The events are stored in following format:
+         * - Each status update is a file, where the name of the file is tracking number + random id.
+         * - Each file has status and time-stamp as the first 2 lines respectively.
+         * - The time at which the file is stored in S3 is not an indication of the time-stamp of the event.
+         * - Once the status is marked as DELIVERED, we can stop tracking the package.
+         * 
+         * A Sample files looks as below:
+         *  FILE-NAME-> '8787323232232332--55322798-dd29-4a04-97f4-93e18feed554'
+         *   >status:IN TRANSIT
+         *   >timestamp: 1573410202
+         *   >Other fields like...tracking history and address
+         */
+        public String handleRequest(ScheduledEvent scheduledEvent, Context context) {
+    
+            final LambdaLogger logger = context.getLogger();
+            try {
+                processShipmentUpdates(logger);
+                return "SUCCESS";
+            } catch (final Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)